src/main/java/org/radarcns/biovotion/BiovotionDeviceManager.java

```diff
@@ -180,6 +225,29 @@ public void start(@NonNull final Set<String> acceptableIds) {
             this.acceptableIds = Strings.containsPatterns(acceptableIds);
             this.isClosed = false;
         }
+
+        // schedule new gap request or checking gap running status
+        gapExecutor.scheduleAtFixedRate(new Runnable() {
+            @Override
+            public void run() {
+                if (isConnected) {
+                    paramReadRequest(VsmConstants.PID_GAP_REQUEST_STATUS);
+                }
+            }
+        }, VsmConstants.GAP_INTERVAL_MS, VsmConstants.GAP_INTERVAL_MS, TimeUnit.MILLISECONDS);
+
+        // schedule new gap request or checking gap running status
+        utcExecutor.scheduleAtFixedRate(new Runnable() {
+            @Override
+            public void run() {
+                if (isConnected) {
+                    // set UTC time
+                    byte[] time_bytes = ByteBuffer.allocate(4).order(ByteOrder.LITTLE_ENDIAN).putInt((int) (System.currentTimeMillis() / 1000d)).array();
+                    final Parameter time = Parameter.fromBytes(VsmConstants.PID_UTC, time_bytes);
+                    paramWriteRequest(time);
+                }
+            }
+        }, VsmConstants.UTC_INTERVAL_MS, VsmConstants.UTC_INTERVAL_MS, TimeUnit.MILLISECONDS);
```

Comment:
Why are these scheduled on different threads? Is execution too slow on a single thread?